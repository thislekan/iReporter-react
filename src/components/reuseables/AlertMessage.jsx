import React from 'react';
import PropTypes from 'prop-types';
import style from '../../styles/modal.css';

function AlertMessage(props) {
  const { openModal, message, resetState } = props;
  return (
    <div>
      {openModal && (
        <div className={style.modal}>
          <div className={style.modal_body}>
            <p>{message}</p>
            <button
              onClick={resetState}
              className={style.modal_btn}
            >
              OK
            </button>
          </div>
        </div>
      )}
    </div>
  );
}

AlertMessage.propTypes = {
  message: PropTypes.string,
  resetState: PropTypes.func,
  openModal: PropTypes.bool,
};

AlertMessage.defaultProps = {
  message: '',
  resetState: f => f,
  openModal: false,
};

export default AlertMessage;
