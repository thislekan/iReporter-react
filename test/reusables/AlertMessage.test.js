import React from 'react';
import { shallow } from 'enzyme';
import AlertMessage from '../../src/components/reuseables/AlertMessage.jsx';

const mockFn = jest.fn();

describe('<AlertMessage />', () => {
  it('Should render the component', () => {
    const wrapper = shallow(
      <AlertMessage
        openModal
        resetState={mockFn}
        message="Hello World"
      />,
    );
    wrapper.find('button').simulate('click', { resetstate: f => f });
    expect(wrapper).toMatchSnapshot();
    expect(mockFn).toHaveBeenCalled();
  });
});
