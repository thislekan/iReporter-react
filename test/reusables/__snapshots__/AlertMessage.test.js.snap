// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertMessage /> Should render the component 1`] = `
<div>
  <div>
    <div>
      <p>
        Hello World
      </p>
      <button
        id="alert-msg-btn"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "resetstate": [Function],
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      >
        OK
      </button>
    </div>
  </div>
</div>
`;
